### PR TITLE
Switch USE_TOKENIZER to opposite NO_TOKENIZER

### PR DESCRIPTION
**Pull request by RyanUnderhill**
Since USE_TOKENIZER is the default, it makes sense to opt out vs opt in when including headers.

This also removes most of the #if statements and instead puts them in one single place and makes it a runtime error that reports "Tokenizer not enabled" if anyone tries to use it.

If we remove this option later, this change makes it even simpler to remove this small piece.

### CODE REVIEW

File: src/models/model.cpp [hunk 1]
```diff
@@ -31,7 +31,18 @@ void State::ClearIO() {
   outputs_.clear();
 }
 
-#if USE_TOKENIZER
+#ifdef NO_TOKENIZER
```
Comment (yufenglee) [inline src/models/model.cpp:34]: #ifndef

File: test/tests.cpp [hunk 2]
```diff
@@ -226,62 +226,4 @@ void Test_BeamSearch_Gpt_Cuda() {
   for (auto model_path : c_tiny_gpt2_model_paths)
     Test_BeamSearch_Gpt_Cuda(model_path.first, model_path.second);
 }
-
-void Test_Phi2_Cuda() {
```
Comment (yufenglee) [inline test/tests.cpp:230]: why do you remove this?

**Review by yufenglee — APPROVED** [review]
:shipit:

File: test/tests.cpp (hunk 2 above)
Comment (RyanUnderhill) [reply test/tests.cpp:230]: It was disabled code that was out of date and never used. I was just cleaning it up when I saw it.

**Pull request merged**